scene/animation/animation_tree.cpp

```diff
@@ -1771,6 +1776,18 @@ void AnimationTree::_notification(int p_what) {
 }
 
 void AnimationTree::set_animation_player(const NodePath &p_player) {
+	AnimationPlayer *new_player = Object::cast_to<AnimationPlayer>(get_node(p_player));
```

Comment:
The problem I see with this is that, if you load the scene, it is unlikely at load time you will be able to resolve this path because the node is not even added to the parent yet. 

I would move this to a separate function `_setup_animation_player`. You can call it from here only if `is_inside_tree()` returns `true`, otherwise you will have to call it from `NOTIFICATION_ENTER_TREE`.